File: greedy/45/my.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:
            return 0
        
        # Initialize variables to track the current max reach and the boundary of current jump
        jumps = 0
        current_max_reach = 0
        current_boundary = 0
        
        # Iterate through the array (except last element as we don't need to jump from there)
        for i in range(len(nums) - 1):
            # Update the farthest position we can reach from current position
            current_max_reach = max(current_max_reach, i + nums[i])
            
            # If we've reached the boundary of current jump
            if i == current_boundary:
                # Make a jump
                jumps += 1
                # Update the boundary to the farthest we can reach
                current_boundary = current_max_reach
                
        return jumps
        

File: dynamic/45/my.c
int jump(int* nums, int numsSize) {
    if (numsSize <= 1) return 0;
    
    int jumps = 0;
    int currentMaxReach = 0;
    int currentBoundary = 0;
    
    for (int i = 0; i < numsSize - 1; i++) {
        currentMaxReach = (currentMaxReach > i + nums[i]) ? currentMaxReach : i + nums[i];
        
        if (i == currentBoundary) {
            jumps++;
            currentBoundary = currentMaxReach;
        }
    }
    
    return jumps;
}